FLT/DedekindDomain/FiniteAdeleRing/BaseChange.lean

```diff
@@ -387,14 +387,33 @@ open scoped TensorProduct -- ⊗ notation for tensor product
 variable [Algebra K (ProdAdicCompletions B L)]
   [IsScalarTower K L (ProdAdicCompletions B L)]
 
+
 noncomputable def ProdAdicCompletions.baseChange :
     ProdAdicCompletions A K →ₐ[K] ProdAdicCompletions B L where
   toFun kv w := (adicCompletionComapAlgHom A K L B _ w rfl (kv (comap A w)))
-  map_one' := sorry -- #232 is this and the next few sorries. There is probably a cleverer way to do this.
-  map_mul' := sorry
-  map_zero' := sorry
-  map_add' := sorry
-  commutes' := sorry
+  map_one' := by
+    simp only
+    refine funext fun w => by rw [Pi.one_apply, Pi.one_apply, map_one]
+  map_mul' := by
+    intro x y
+    simp only
+    refine funext fun w => by rw [Pi.mul_apply, Pi.mul_apply, map_mul]
+  map_zero' := by
+    simp only
+    refine funext fun w => by rw [Pi.zero_apply, Pi.zero_apply, map_zero]
+  map_add' := by
+    intro x y
+    simp only
+    funext w
+    haveI : Module K (adicCompletion L w) := Algebra.toModule
```

Comment:
That is weird that you need this. It should really be `letI` though because it's data not a proof. I'm not too bothered about it though as it's just in a proof.

Reply:
I was surprised by this too. I went ahead and changed it to letI.